src/engine.rs

```diff
@@ -1,18 +1,60 @@
 use anyhow::{anyhow, Result};
+use rust_embed::RustEmbed;
 use std::{
+    collections::HashSet,
     io::Cursor,
     path::PathBuf,
     time::{Duration, Instant},
 };
-use wasmtime::{Engine, Linker, Module, Store};
+use wasi_common::WasiCtx;
+use wasmtime::{Config, Engine, Linker, Module, Store};
 
 use crate::function_run_result::{
     FunctionOutput::{self, InvalidJsonOutput, JsonOutput},
     FunctionRunResult, InvalidOutput,
 };
 
+#[derive(RustEmbed)]
+#[folder = "providers/"]
+struct StandardProviders;
+
+fn import_modules(
+    module: &Module,
+    engine: Engine,
+    linker: &mut Linker<WasiCtx>,
+    mut store: &mut Store<WasiCtx>,
+) {
+    let imported_modules: HashSet<String> =
+        module.imports().map(|i| i.module().to_string()).collect();
+    imported_modules.iter().for_each(|imported_module| {
+        let mut imported_module = imported_module.to_owned();
+        imported_module.push_str(".wasm");
+
+        let imported_module_bytes = StandardProviders::get(&imported_module);
+
+        if let Some(bytes) = imported_module_bytes {
```

Comment:
Should we have an expectation above, rather than using `if let` here? Seems like it would be bad for this to silently fail.

Reply:
I left this in to handle cases where something is imported that we don't have available. The wasi context is already linked and is one example where it is included in the imports. This is already linked.